cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -480,7 +481,33 @@ fn lower_to_amode<C: LowerCtx<I = Inst>>(ctx: &mut C, spec: InsnInput, offset: i
             )
         } else {
             for i in 0..=1 {
-                if let Some(cst) = ctx.get_input(add, i).constant {
+                let input = ctx.get_input(add, i);
+
+                // Try to pierce through uextend.
+                if let Some(uextend) = matches_input(
+                    ctx,
+                    InsnInput {
+                        insn: add,
+                        input: i,
+                    },
+                    Opcode::Uextend,
+                ) {
+                    if let Some(cst) = ctx.get_input(uextend, 0).constant {
+                        // Zero the upper 32 bits.
+                        let input_size = ctx.input_ty(uextend, 0).bits();
+                        let shift = 64 - input_size;
+                        let uext_cst = (cst << shift) >> shift;
```

Comment:
I'm sure this is correct, but .. nevertheless .. it's hard to reason about the correctness of it (in particular, the expression `(cst << shift) >> shift`, given that there there is no type information here.  I'm assuming that `cst` is a `u64` here.  Could you please add some type annotation (or something else, that the compiler can check) so as to make that clear to the reader?

Reply:
Yes, it is an u64 (so the right shift is logical); i'll add an explicit type annotation for uext_cst and shift!